kolibri/core/assets/src/views/ContentRenderer/__tests__/DownloadButton.spec.js

```diff
@@ -0,0 +1,80 @@
+import Vue from 'vue';
+import { render, screen } from '@testing-library/vue';
+import VueRouter from 'vue-router';
+import useUser, { useUserMock } from 'kolibri.coreVue.composables.useUser';
+import DownloadButton from '../DownloadButton.vue';
+import { RENDERER_SUFFIX } from '../constants';
+import '@testing-library/jest-dom';
+
+jest.mock('kolibri.coreVue.composables.useUser');
+
+const getDownloadableFile = (isExercise = false) => {
+  const PRESET = isExercise ? 'exercise' : 'thumbnail';
+
+  // Register a component with the preset name so that the file is considered renderable
+  Vue.component(PRESET + RENDERER_SUFFIX, { template: '<div></div>' });
+
+  return {
+    preset: PRESET,
+    available: true,
+    file_size: 100,
+    storage_url: 'http://example.com/sample.png',
+    extension: 'png',
+    checksum: '1234567890',
+  };
+};
+
+// A helper function to render the component with the given props and some default mocks
+const renderComponent = props => {
+  useUser.mockImplementation(() =>
+    useUserMock({
+      isAppContext: props.isAppContext || false,
+    })
+  );
+
+  return render(DownloadButton, {
+    routes: new VueRouter(),
+    props: {
+      files: [],
+      nodeTitle: '',
+      ...props,
```

Comment:
Just flagging that we are still passing the `isAppContext` as prop here. To prevent this from happening, we can use destructuring:

```javascript
const renderComponent = props => {
  const { isAppContext, ...componentProps } = props;
  ...
}; 
```

Reply:
Got it! Since how we are using the name `useUserMock` for all the composable mocks, the structure would be a bit different, but I get the spirit. Thanks!